docs/manual/resource-leak-checker.tex

```diff
@@ -0,0 +1,343 @@
+\htmlhr
+\chapterAndLabel{Resource Leak Checker for must-call obligations}{resource-leak-checker}
+
+The Resource Leak Checker guarantees that the program fulfills every object's
+must-call obligations before the object is de-allocated.
+
+A resource leak occurs when a program does not explicitly dispose of some finite
+underlying resource, such as a socket, file descriptor, or database connection.  To dispose
+of the resource, the program should call some method on an object.
+(De-allocating or garbage-collecting the object is not sufficient.)  For
+example, the program must call \<close()> on every object that implements the
+interface \<java.io.Closeable>.
+
+The Resource Leak Checker can check any property of the form ``the programmer
+must call all methods \emph{M} on object \emph{O} before \emph{O} is
+de-allocated''.  For resource leaks, by default \emph{M} is the set containing
+\<close()> and \emph{O} is any object that implements the interface
+\<java.io.Closeable>.  You can extend this guarantee to other types and methods
+by writing \<@MustCall> or \<@InheritableMustCall> annotations, as described in
+Section~\ref{must-call-annotations}.
+
+The Resource Leak Checker works in three stages:
+\begin{enumerate}
+\item The Must Call Checker (\chapterpageref{must-call-checker})
+  over-approximates each expression's must-call methods as a
+  \refqualclass{checker/mustcall/qual}{MustCall} type.
+\item The Called Methods Checker (\chapterpageref{called-methods-checker})
+  under-approximates each expression's definitely-called methods as a
+  \refqualclass{checker/calledmethods/qual}{CalledMethods} type.
+\item When any program element goes out of scope (i.e., it is ready to be
+  de-allocated), the Resource Leak Checker compares the types
+  \<@MustCall(\emph{MC})> and \<@CalledMethods(\emph{CM})>.  It reports an error
+  if there exists some method in \emph{MC} that is not in \emph{CM}.
+\end{enumerate}
+
+
+\sectionAndLabel{How to run the Resource Leak Checker}{resource-leak-run-checker}
+
+\begin{Verbatim}
+javac -processor ResourceLeak MyFile.java ...
```

Comment:
FYI I just tried running with `-processor ResourceLeak` and it didn't work.  I had to use `-processor org.checkerframework.checker.resourceleak.ResourceLeakChecker`

Reply:
Could you try `-processor resourceleak` (all lower case)?

Reply:
`-processor resourceleak` does not work either.  This is within a Maven build, with the Checker artifact in the annotation processor path and passing `-processor` and `resourceleak` within `<arg>` elements.

Reply:
FWIW `-processor resourceleak` works when running with `checker/bin/javac`

Reply:
That's what I was thinking of.  The short forms only work when running `CheckerMain`.  We could put the long version here, or show both versions -- the short lowercase version and the long fully-qualified one.

Reply:
I think showing both would be quite useful.  It's happened before that I wanted to run a checker via a build system, tried the short form based on the manual, and then had to dig out the long form from the source code

Reply:
I think the short syntax only works when using the Checker Framework's `javac` wrapper, which you're probably not using if you're in a Maven build. Other checkers have similar text in their manual sections that will suffer from the same problem, so if we think this is confusing here we should fix it throughout the manual.

Reply:
I would be in favor of fixing throughout yes

Reply:
I didn't find many occurrences. #4717 makes the few changes that are necessary (unless I missed some).